app/sign.js

```diff
@@ -192,8 +192,13 @@ const handleSignXrp = function(recoveryRequest, key, skipConfirm) {
   const rippleKeypairs = require('ripple-keypairs');
   const rippleParse = require('ripple-binary-codec');
 
+  // handle different input formats
+  if (!recoveryRequest.txHex) {
```

Comment:
wait, so this fcn can be called with a request that either has:

1. `tx`
2. `txHex` and/or 
3. `transactionHex` 

defined on it? Are these values ever the same? When/why would they differ? When/why would some attrs be null and some be populated with values?

Can we force the caller to pass an explicit type of data with an explicit key?

Reply:
these all refer to the transaction hex. sometimes we get recovery request files from the BitGo WWW, sometimes from the Wallet Recovery Wizard, and sometimes from the KRS provider. So far I've seen the transaction hex be called tx, txHex, or transactionHex in these recovery files. it would be nice to standardize

Reply:
> . So far I've seen the transaction hex be called tx, txHex, or transactionHex in these recovery files. 

Is that the same for all recovery transactions, or just Ripple transactions?

.......

If it's for everything, would it make sense to create a generic util fcn like: 

```javascript
const normalizeTransactionHex = function(recoveryRequest){
  if( recoveryRequest.tx ){
    return recoveryRequest.tx
  }
  if (recoveryRequest.txHex ){
    return recoveryRequest.txHex
  }
  if (recoveryRequest.transactionHex ){
    return recoveryRequest.transactionHex
  }
   
  throw("The recovery request did not provide a transaction hex");
}
```

and call that fcn whenever we are sanitizing the recovery request.